core/embed/trezorhal/stm32u5/haptic/drv2625/drv2625.c

```diff
@@ -78,50 +77,55 @@ static haptic_driver_t g_haptic_driver = {
     .initialized = false,
 };
 
-static bool drv2625_set_reg(uint8_t addr, uint8_t value) {
+static ts_t drv2625_set_reg(uint8_t addr, uint8_t value) {
+  verify_init();
+
   uint8_t data[] = {addr, value};
-  return i2c_transmit(DRV2625_I2C_INSTANCE, DRV2625_I2C_ADDRESS, data,
-                      sizeof(data), 1) == HAL_OK;
+
+  HAL_StatusTypeDef hal_status = i2c_transmit(
+      DRV2625_I2C_INSTANCE, DRV2625_I2C_ADDRESS, data, sizeof(data), 1);
+
+  verify_hal_ok(hal_status);
+
+error:
+  return verify_status();
 }
 
-bool haptic_init(void) {
+ts_t haptic_init(void) {
+  verify_init();
+
   haptic_driver_t *driver = &g_haptic_driver;
 
-  if (driver->initialized) {
-    return false;
-  }
+  verify(driver->initialized, TS_ERROR_NOTINIT);
 
   memset(driver, 0, sizeof(haptic_driver_t));
 
+  ts_t status;
+
   // select library
-  if (!drv2625_set_reg(DRV2625_REG_LIBRARY,
-                       LIB_SEL | DRV2625_REG_LIBRARY_GAIN_25)) {
-    return false;
-  }
+  status = drv2625_set_reg(DRV2625_REG_LIBRARY,
+                           LIB_SEL | DRV2625_REG_LIBRARY_GAIN_25);
+  verify_ok(status);
```

Comment:
What's the use of the `status` variable here?

Reply:
If you split this into two lines, you can place a breakpoint at the line containing `verify_ok()` to observe the operation status. This pattern is very useful.